crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -256,6 +256,29 @@ impl TransactionEffectsAPI for TransactionEffectsV2 {
             .collect()
     }
 
+    fn transferred_from_consensus(&self) -> Vec<ObjectRef> {
+        self.changed_objects
+            .iter()
+            .filter_map(|(id, change)| {
+                match (
+                    &change.input_state,
+                    &change.output_state,
+                    &change.id_operation,
+                ) {
+                    (
+                        ObjectIn::Exist((_, Owner::ConsensusV2 { .. })),
+                        ObjectOut::ObjectWrite((
+                            object_digest,
+                            Owner::AddressOwner(_) | Owner::ObjectOwner(_) | Owner::Immutable,
+                        )),
+                        IDOperation::None,
+                    ) => Some((*id, self.lamport_version, *object_digest)),
+                    _ => None,
```

Comment:
consider 

```
match (
     &change.input_state,
     &change.output_state,
     &change.id_operation,
 ) {
     (
         ObjectIn::Exist((_, Owner::ConsensusV2 { .. })),
         ObjectOut::ObjectWrite((object_digest, Owner::ConsensusV2 { .. })),
         IDOperation::None,
     ) => None,
     (
         ObjectIn::Exist((_, Owner::ConsensusV2 { .. })),
         ObjectOut::ObjectWrite((object_digest, _)),
         IDOperation::None,
     ) => Some((*id, self.lamport_version, *object_digest)),
     _ => None,
 }
```

as a more future proof alternative

Reply:
Can you elaborate on how this is more future-proof? If we introduce a new Owner type, there's no way of knowing in advance whether it will be a consensus type or a fastpath type.